services/billing.py

```diff
@@ -155,18 +157,75 @@ def delete_subscription(self, owner: Owner):
             f"Downgrade to basic plan from user plan for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
             extra=dict(ownerid=owner.ownerid),
         )
+
         if subscription_schedule_id:
             log.info(
                 f"Releasing subscription from schedule for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
                 extra=dict(ownerid=owner.ownerid),
             )
             stripe.SubscriptionSchedule.release(subscription_schedule_id)
 
-        stripe.Subscription.modify(
-            owner.stripe_subscription_id,
-            cancel_at_period_end=True,
-            proration_behavior="none",
+        # we give an auto-refund grace period of 24 hours for a monthly subscription or 72 hours for a yearly subscription
+        current_subscription_datetime = datetime.fromtimestamp(
+            subscription["current_period_start"]
+        )
+        difference_from_now = datetime.now() - current_subscription_datetime
+
+        subscription_plan_interval = getattr(
+            getattr(subscription, "plan", None), "interval", None
         )
+        within_refund_grace_period = (
+            subscription_plan_interval == "month" and difference_from_now.days < 1
+        ) or (subscription_plan_interval == "year" and difference_from_now.days < 3)
+
+        customer = stripe.Customer.retrieve(owner.stripe_customer_id)
+        # we are giving customers 2 autorefund instances
+        autorefunds_remaining = int(
+            customer["metadata"].get("autorefunds_remaining", "2")
+        )
```

Comment:
I want this logic to be inside the `if within_refund_grace_period` block so it's only called when they are within the cancellation window

Reply:
That makes sense. Any ideas about how to prevent another level of `if` statements or in your opinion, does that seem fine? Do we have a general rule of thumb for how many levels we think is acceptable?